src/xpcc/architecture/platform/driver/core/cortex/driver.xml

```diff
@@ -38,11 +39,22 @@
 		<!-- stm32f7 -->
 		<template device-family="f7" out="linkerscript.ld">stm32/stm32_idtcm.ld.in</template>
 
-		<template>startup.c.in</template>
-		<template>delay.cpp.in</template>
-		<template>syscalls.cpp.in</template>
+		<!-- dealing with C++ -->
 		<static>cxxabi.cpp</static>
+		<!-- dealing with newlib -->
+		<static>syscalls.c</static>
+		<!-- startup script -->
+		<template>startup.c.in</template>
 		<static>default_handler.sx</static>
+		<!-- everything to do with dynamic memory -->
+		<static>heap_table.c</static>
+		<template>heap_newlib.c.in</template>
```

Comment:
Ok, I'm really not opposed to using CMake, I did integrate Jinja2 templating into CMake before and it wasn't totally terrible.

I'm not sure if we need IDE support for the HAL templates though. As @dergraaf pointed out some time ago, the HAL generation should really be a separate step and the resulting HAL should then be build together with the rest of the framework.
I would therefore favour using Python for the HAL generation part, with a CMake wrapper of course. Library builder is probably the better choice than Conan, due to being written with exactly this in mind.